src/Engine/LocalizeDictionary.cs

```diff
@@ -979,22 +979,15 @@ internal static class DictionaryEvent
             /// <param name="args">The event arguments.</param>
             internal static void Invoke(DependencyObject sender, DictionaryEventArgs args)
             {
-                var list = new List<IDictionaryEventListener>();
-
                 lock (ListenersLock)
                 {
-                    foreach (var wr in Listeners.ToList())
+                    foreach (var listener in Listeners.GetListeners())
                     {
-                        var targetReference = wr.Target;
-                        if (targetReference != null)
-                            list.Add((IDictionaryEventListener)targetReference);
-                        else
-                            Listeners.Remove(wr);
+                        listener.ResourceChanged(sender, args);
```

Comment:
if you already create that ListnersClass why not even integrate fully that loop as a method like:
Listerns.ResourceChanged(sender,args) and than this is fully handled in the ListnersClass

Reply:
I've decided `ListenerList` should provide only base methods - add, remove and enumerate without additional logic. I can integrate this method inside if it is necessary. But what should I do with exceptions? It seems to me that better always call `ResourceChanged` for all listeners.

Reply:
You are right. Exceptions is a good argument

Reply:
Sorry, did I understand correctly? I don't move this method to 'ListenersList`